packages/starknet-snap/src/utils/starknetUtils.ts

```diff
@@ -1079,7 +1075,7 @@ export async function getEstimatedFees(
     suggestedMaxFee: estimateFeeResp.suggestedMaxFee.toString(10),
     overallFee: estimateFeeResp.overall_fee.toString(10),
     unit:
-      transactionVersion === constants.TRANSACTION_VERSION.V2
+      invocationsDetails.version === constants.TRANSACTION_VERSION.V1
```

Comment:
```suggestion
      invocationsDetails?.version === constants.TRANSACTION_VERSION.V3 ? FeeTokenUnit.STRK: FeeTokenUnit.ETH
```

it is better to if it is v3, then strk, everything else eth